addon/adapters/pouch.js

```diff
@@ -363,10 +359,33 @@ export default DS.RESTAdapter.extend({
     return this.get('db').rel.save(this.getRecordTypeName(type), data);
   },
 
-  deleteRecord: function (store, type, record) {
+  deleteRecord: function (store, type, snapshot) {
     this._init(store, type);
-    var data = this._recordToData(store, type, record);
-    return this.get('db').rel.del(this.getRecordTypeName(type), data)
-      .then(extractDeleteRecord);
+    let data = this._recordToData(store, type, snapshot);
+
+    //Delete the children
+    let deletedChildren = [];
+    snapshot.eachRelationship(function(name, descriptor) {
+      if(descriptor.kind === 'hasMany' && descriptor.options['cascadeDestroy']) {
+        snapshot.record.get(name).then(function(childRecords) {
+          childRecords.forEach(function(childRecord) {
```

Comment:
I feel like you could use `map` here, to map the `destroyRecord()` instead of using `forEach`.

```
childRecords.map((item) => item.destroyRecord());
```

[The former will be faster than both `forEach` **and** `for`](https://ryanpcmcquen.org/javascript/2015/10/25/map-vs-foreach-vs-for.html).